lib/kubernetes-deploy/kubernetes_resource/redis.rb

```diff
@@ -7,8 +7,10 @@ class Redis < KubernetesResource
     SYNC_DEPENDENCIES = %w(Deployment Service)
     def sync(mediator)
       super
-      @deployment = mediator.get_instance(Deployment.kind, "redis-#{redis_resource_uuid}")
-      @service = mediator.get_instance(Service.kind, "redis-#{redis_resource_uuid}")
+      @deployment = mediator.get_instance(Deployment.kind, name) ||
+        mediator.get_instance(Deployment.kind, deprecated_name)
```

Comment:
why not deprecated_name || new_name? I guess you want the new name to have precedence?